src/effects/effectchain.cpp

```diff
@@ -216,7 +216,20 @@ void EffectChain::loadChainPreset(EffectChainPresetPointer pPreset) {
     setMixMode(pPreset->mixMode());
     m_pControlChainSuperParameter->setDefaultValue(pPreset->superKnob());
 
-    m_presetName = pPreset->name();
+    // If this is a regular, editable effect chain (i.e. not a QuickEffectChain)
+    // and the empty '---' preset was selected, set an empty name in order to
+    // * clear WEffectChainPresetSelector with chainPresetChanged() (index will be -1)
+    // * not store '---' in effects.xml (e.g. because the respective
+    //   empty '---' preset is not going to be stored either)
+    // With QuickEffect chains this doesn't matter because only the name will be
+    // stored for each deck.
+    // For details check EffectsManager::saveEffectsXml()
+    if (pPreset->name() == kNoEffectString &&
+            !m_pEffectsManager->chainIsQuickEffectChain(this)) {
+        m_presetName = QString();
+    } else {
+        m_presetName = pPreset->name();
+    }
```

Comment:
No, we can't.
This is required to deal with how QuickEffect vs. regular chain presets are stored.
It is the least invasive solution, and it works perfectly for both chain types.

I try to explain it once more, this is best understood if you look at your effects.xml.

* **QuickEffect chains** are not editable, only super knob and enable switch are exposed¹
* for Quick Effect chains only the preset name is stored in effects.xml
* on next start, this name is looked up in the comon preset list
* if this field is empty or preset not found, the default preset is loaded (Filter)
* = `---` will accomplish that the empty chain is restored correctly

__
* for **regular chains** (e.g. EffectRack1) the entire chains state is stored
* this maybe an unchanged default preset, or some custom chain that wasn't even saved or exported, yet, so the preset name is optional
* the empty `---` preset itself _must not be edited_, it is just a shortcut to clear the entire chain and allow the user to start over
= eject preset = clear slots & preset name = the user can immediatley start to create a new chain
* thus, if the chain would adopt the `---` as preset name, that would be incorrect as soon as an effect slot is populated
* the only way to save/overwrite any effect chain preset is via WEffectChainPresetButton:
when `effectChainPresetListUpdated` or `chainPresetChanged` are received [the menu will be populated](https://github.com/mixxxdj/mixxx/pull/10859/files#diff-be1dd872669ac0a1d01711568bbb7890d5288f7c198e80f44dfd7bb4362a264fR50), and since (empty name) will not be found in the common preset list, the preset is considered null and "Update Preset" is not shown (same applies to [`WEffectChainPresetSelector`](https://github.com/mixxxdj/mixxx/pull/10859/files#diff-2ac3fea593519ba78388b958c24246fd446bd7fb86f4c2c1d067699790c34d53R101)

¹ note that this is only true with official skins. IIUC it's not prohibited in any way to create WEffectChainPresetButton and WEffectSeletor widgets for a QuickEffect chain, so with those one could alter the `---` preset, though that state would not be saved automatically and thus not be restored on next start.
Idk if we need to add a guard to not allow this in the first place to avoid disaapointment with custom skins that implemet those widgets.

Reply:
I updated the above description. Hopefully it's comprehensible ths time.

Reply:
I have tested the LatNight effect chain selector and label and I have noticed that WEffectChain is cleared instead of dashed out. Is this desired, or only a side effect of the this code here? 
For my understanding it should be "---" instead and the code above needs to be removed. 

The solution with 
m_pEffectsManager->chainIsQuickEffectChain(this)
is also a bit unusual because now we have QuickEffectChain code in the base class even though QuickEffectChain has already an implementation of loadChainPreset() 

I have just did a test with replacing the code above with 
`m_presetName = pPreset->name();`
And it works IMHO flawlessly after deleting the assertion in EffectChainPresetManager::saveEffectsXml() which is IMHO wrong, because it saves the current state, independent from which chain was loaded. 

This way the preset name is "---" but the changes after selecting "---" are still recorded, which is valid and happens with any other chain. 

If you want to reflect that the chain is no longer empty, we may rename it in "New Chain". This however happens with any other chain you pick and than change the effect. For instance if you Pick the Reverb preset and replace the Reverb effect with a bit-chusher you have the same inconsistent state. 

> the empty --- preset itself must not be edited,

Why not, like any other preset. It must not be stored as preset though, but that is already disabled elsewhere. 

> thus, if the chain would adopt the --- as preset name,

This is not a problem of this PR, because it happens with any other preset in the same way. 
If we want to fix that, it is a topic of another PR. 

> = selecting another preset, then going back to --- would not clear the chain!

This works in my test. 

Maybe I am still missing something?

Reply:
> The solution with
m_pEffectsManager->chainIsQuickEffectChain(this)
is also a bit unusual because now we have QuickEffectChain code in the base class even though QuickEffectChain has already an implementation of loadChainPreset()

If chosen this path to avoid code duplication. If you think this needs to be resolved I can move it the quickeffectchain.cpp

Reply:
The basic rule I want to adhere to is that `---` _always_ means "empty" in any other selector in skins and preferences.
I built everything on this to find a way to clear the QuickEffect chains.

It gets tricky only with standart effect chains, because IMO for those the `---` item should be an **action** not a **state**, it should allow the user to restore the initial state (empty chain) like after starting Mixxx the first time.
Yes, the inconsistency of keeping `---` (or any other preset name) after changing an effect applies to all preset, though IMO it is hard to distinguish if a user alters _EchoVerbHP_ by selecting a different filter or adds a entirely different effect. But we should accept this inconsistency for the very special `---` item.

Btw I just notice a somewhat big mistake: actually the chain preset selectors are not part of official skins (and neither are WEffectChain labels) so `---` should be in the menu of WEffectChainPresetButton (and not be marked selected after selecting it). Or we add a separate "Clear chain" item. I think `---` would be more intuitive, though I have no strong preference.

Reply:
> > = selecting another preset, then going back to --- would not clear the chain!

> This works in my test.

True, it works, even when `---` is already selected. But  definitely expereinced that, probably with an early state of this branch, maybe in the preset menu, idk.

Reply:
> But we should not accept this inconsistency for the very special --- item.

Accepting it would make the behavior consistent that the chain label only displays the last **action**. Special case the "---" case and replace it with "" does not solve the underlying issue, that you expectation is to display the **state**. 
"" is also shown in case the chain is still "---" which feels wrong as well. 

This can be solved by removing the label once the user has re-added an effect, or in case of other preset once they have changes any effect. For this PR I would prefer to keep the last **action** display feature working for all.

Reply:
> I would rather change that, instead of undoing, half of this good working PR.

Oh, it's little work actually. Just not adding it to the standard chains makes the special treatment obsolete in a few places, and I can remove `chainIsQuickeffectChain`.

> Apart from these few renaming lines, the PR is well designed and I like that we treat all chains the same.

Hmm, as our discussion unveils that's exactly the issue, because there GUI representation is not the same. QuickEffect chains have a stripped GUI and their preset selectors look like the single effec slot selectors now, and that's fine. Standard chains have the full GUI and maybe even don't necessarily need `---` or "Clear chain". Thus I'm fine with making a few steps backwards.